runtime/doc/develop.txt

```diff
@@ -166,77 +166,243 @@ stick to these rules, to keep the sources readable and maintainable.
+<
+Wrong: >
+    if (cond)
 	/*
-	 * The "foo" argument does something useful.
-	 * Return OK or FAIL.
+	 * comment
 	 */
-For new code or lines of code that change, it is preferred to use // comments.
-Especially when it comes after code:
-	int some_var;  // single line comment useful here
+	cmd;
+    else
+	cmd;
+<
+When an `if`/`else` has braces on one block, the other should have it too.
+OK: >
+    if (cond)
+    {
+	cmd;
+    }
+    else
+    {
+	cmd;
+	cmd;
+    }
+<
+Wrong: >
+    if (cond)
+	cmd;
+    else
+    {
+	cmd;
+	cmd;
+    }
+
+    if (cond)
+    {
+	cmd;
+	cmd;
+    }
+    else
+	cmd;
+<
+OK: >
+    while (cond)
+	cmd;
+<
+Wrong:
+>
+    while (cond)
+	if (cond)
+	    cmd;
+<
+
+
+TYPES							    *style-types*
+
+Use descriptive types. You can find a list of them in the src/structs.h file
+and probably in a typedef in the file you are working on.
+
+Note that all custom types are postfixed with "_T"
+
+OK: >
+    int is_valid_line_number(linenr_T lnum);
+<
+Wrong: >
+    int is_valid_line_number(unsigned long lnum);
+<
+
+
+SPACES AND PUNCTUATION					   *style-spaces*
+
+No space between a function name and the bracket:
+
+OK:	func(arg);
+Wrong:  func (arg);
+
+Do use a space after `if`, `while`, `switch`, etc.
+
+OK:	if (arg)	for (;;)
+Wrong:	if(arg)		for(;;)
 
-Enums ~
+Use a space after a comma or semicolon:
 
-The last item in an enum may have a trailing comma.  C89 didn't allow this.
+OK:	func(arg1, arg2);	for (i = 0; i < 2; ++i)
+Wrong:  func(arg1,arg2);	for (i = 0;i < 2;++i)
 
-Types ~
+Use a space before and after '=', '+', '/', etc.
 
-"long long" is allowed and can be expected to be 64 bits.  Use %lld in printf
-formats.  Also "long long unsigned" with %llu.
+Wrong:	var=a*5;
+OK:	var = a * 5;
 
-Declarations ~
+Use empty lines to group similar actions together.
 
-Now that the minimal supported compiler is MSVC 2015 declarations do not need
-to be at the start of a block.  However, it is often a good idea to do this
-anyway.
+OK: >
+    msg_puts_title(_("\n--- Signs ---"));
+    msg_putchar('\n');
 
-Declaration of the for loop variable inside the loop is recommended:
-	for (int i = 0; i < len; ++i)
-Since this is clearly an advantage we'll use this more often.
+    if (rbuf == NULL)
+	buf = firstbuf;
+    else
+	buf = rbuf;
 
+    while (buf != NULL && !got_int)
+<
+Wrong: >
+    msg_puts_title(_("\n--- Signs ---"));
+    msg_putchar('\n');
+    if (rbuf == NULL)
+	buf = firstbuf;
+    else
+	buf = rbuf;
+    while (buf != NULL && !got_int)
+<
 
-Not to be used ~
 
-These C99 features are not to be used, because not enough compilers support
-them:
-- Variable length arrays (even in C11 this is an optional feature).
-- _Bool and _Complex types.
-- "inline" (it's hardly ever needed, let the optimizer do its work)
-- flexible array members: Not supported by HP-UX C compiler (John Marriott)
+FUNCTIONS						*style-functions*
+
+Use function declarations with the return type on a separate indented line.
+
+OK: >
+    int
```

Comment:
this should be indented, no?

Reply:
I've checked some formatters and none of them support this kind of formatting. But, having said that, I can make a PR to https://github.com/uncrustify/uncrustify to support this if it is a MUST.

Reply:
Okay, I am just used to that style. It's not a must for me. More important, that function definitions start at column 1.